docs/decisions/0002-limiting-stream-length.rst

```diff
@@ -0,0 +1,52 @@
+0001 Limiting stream length
+###########################
+
+Status
+******
+
+**Provisional**
+
+Context
+*******
+
+The redis streams are an append-only data structure that stores entries
+forever. However storing these events forever is neither required nor it is
+efficient to store it in redis memory.
+
+Decision
+********
+
+There are two options to cap these streams to a specific length.
+
+Running `XTRIM`_ command regularly by creating custom django command. It
+supports two strategies to evict older entries.
+
+.. _XTRIM: https://redis.io/commands/xtrim/
+
+
+* MAXLEN: Evicts entries as long as the stream's length exceeds the specified
+  threshold, where threshold is a positive integer.
+* MINID: Evicts entries with IDs lower than threshold, where threshold is a
+  stream ID.
+
+
+Another option which only provides single strategy i.e. ``MAXLEN`` but
+simpler in terms of implementation as well as maintenance is including maximum
+length of stream while adding items to the stream.
+
+.. code-block::
+
+   stream.add(data, id='*', maxlen=None, approximate=True)
+
+We can set maxlen value via django settings with a default value of 1000.
```

Comment:
@navinkarkera Do we have any stats to back up the decision for picking this value? E.g.:
* How many events are triggered on a arbitrary system?
* Which variables are likely to influence the rate (e.g. number of users)?

If not, I suggest adding a note that this value is guesstimated.

Also, I'm not sure what would happen if the rate at which the events are emitted is higher than the services subscribed can process them. Is this not a valid concern for the use case?

Reply:
@Cup0fCoffee afaik we don't have any stats and this is just an guesstimate, also the number of events triggered might be different for different users so the default can only be a guess but it will be configurable. I'll add a note as you suggested.

>Also, I'm not sure what would happen if the rate at which the events are emitted is higher than the services subscribed can process them. Is this not a valid concern for the use case?

I am not sure if I understand, there should be any issue if the rate of events emitted is higher than the consumers processing them.